services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +46,46 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  BaseType_t send = xQueueSend(thermalMgrQueueHandle, event, 0); // 0 ticks to wait 
+  if (send != pdPASS) return ERR_CODE_INVALID_QUEUE_MSG/*failed case*/;
   return ERR_CODE_SUCCESS;
 }
 
-void osHandlerLM75BD(void) {
+void osHandlerLM75BD(void) { 
   /* Implement this function */
+  float* temperature;
+  // Question: a, I using the correct devAddr 
+    error_code_t errCode = readTempLM75BD(LM75BD_OBC_I2C_ADDR, temperature); // Use your readTempLM75BD function
```

Comment:
since you are declaring temperature to be a pointer to a float but not setting it to anything, when you pass it into this function it is a NULL pointer so readTemp will return an error